Future.ts

```diff
@@ -43,7 +50,7 @@ export default class Future<L, R> {
      * @return {Promise<R>} Start execution of the Future but return a Promise which will be resolved/reject when the Future is
      */
     toPromise(): Promise<R> {
-        return new Promise<R>((resolve: Resolve<R>, reject: Reject<L>) => this.engage(reject, resolve));
+        return new Promise<R>((resolve: Resolve<R>, reject: Reject<L>) => this.engageAndCatch(reject, resolve));
```

Comment:
I think this can be `engage`? A throw from inside a Promise like this should get caught by the promise. Actually why does `engageAndCatch` exist with the `safe` flag like this at all? If we want it, can't we have `noThrowEngage` or `engageAndCatch` without the `safe` argument and always do the `safe` logic? Anywhere you would write `engageAndCatch(x, y, false)` you can write `engage(x, y)` instead right?

Reply:
You're right on both accounts. Fixed and added a test that shows that an error thrown in the future will be caught by the promise if in a chain.